JockeyJS/js/jockey.js

```diff
@@ -187,14 +238,12 @@
     // be shown that says "Safari cannot open page because the URL is invalid."
     // From here: http://stackoverflow.com/questions/4460205/detect-ipad-iphone-webview-via-javascript
 
-    var UIWebView = /(iPhone|iPod|iPad).*AppleWebKit(?!.*Safari)/i.test(navigator.userAgent);
-	var isAndroid = navigator.userAgent.toLowerCase().indexOf("android") > -1;
-	
-    if ((iOS && UIWebView) || isAndroid) { 
-        Jockey.dispatcher = Dispatcher;
-    } else {
-        Jockey.dispatcher = nullDispatcher;
+    var UIWebView  = /(iPhone|iPod|iPad).*AppleWebKit(?!.*Safari)/i.test(navigator.userAgent);
+    var isAndroid  = navigator.userAgent.toLowerCase().indexOf("android") > -1;
+
+    if ((iOS && UIWebView) || isAndroid) {
+        Jockey.dispatchers.push(nativeDispatcher);
```

Comment:
nativeDispatcher + iframeDispatcher won't work on iphone (which don't have the native jockey lib)